site/blog/2022-10-28-optimize-s3-endpoints.mdx

```diff
@@ -0,0 +1,126 @@
+    <button class="button button--lg button--primary">Add missing interface endpoint</button>
+  </a>
+</p>
+
+By clicking on the button, our engine will auto-generate the missing interface endpoints and will allow you
+to preview the changes to be applied on your cloud. <a href="https://docs.aws.amazon.com/vpc/latest/privatelink/create-interface-endpoint.html">Get more detail about interface endpoints.</a>
+
+## Endpoint gateways
+
+And endpoint gateway is a gateway that can be configured as a target for a route in your route table, used
+to access traffic in DynamoDB or S3
+
+Multiple gateway endpoints can be created in a single VPC, and those can be used on different route tables
+to enforce different access policies from different subnets to the same service.
+
+Gateway endpoints are supported within the same region only, resources from multiple regions cannot be
+accssed, even using VPC peering. They also support IPv4 traffic only.
+
+In order to use them, DNS resolution must be enabled in the VPC.
+
+When that route is added, all instances in the subnets associated with the route table will automatically
+use the endpoint to access the service.
+
+Gateway endpoints only support S3 and DynamoDB services.
+
+<a href="https://app.iasql.com/#/button/Add S3 interface endpoints/--%20Inserts%20the%20missing%20endpoints%0Ainsert%20into%20endpoint_gateway%28region%2C%20vpc_id%2C%20service%29%20select%20bucket.region%2C%20vpc.id%2C%20%27s3%27%20from%20bucket%20inner%20join%20vpc%20on%20bucket.region%3Dvpc.region%20where%20not%20exists%20%28select%20id%20from%20endpoint_gateway%20where%20endpoint_gateway.region%3Dbucket.region%20and%20endpoint_gateway.vpc_id%3Dvpc.id%29%0A--%20Preview%20the%20changes%0Aselect%20%2A%20from%20iasql_preview_apply%28%29%3B%0A--%20Apply%20the%20changes%0A--select%20%2A%20from%20iasql_apply%28%29%3B">
+  <button class="button button--lg button--primary">Add missing endpoints gateways</button>
+</a>
+
+By clicking on the button, our engine will auto-generate the missing endpoint gateways and will allow you to
+preview the changes to be applied on your cloud. <a href="https://docs.aws.amazon.com/vpc/latest/privatelink/gateway-endpoints.html">Get more detail about gateway endpoints.</a>
+
+## Testing the result
+
+Once all the relevant endpoints have been created, please check your access from internal EC2 instances or
+other resources - please use a VM on the same region as the endpoint you want to test:
```

Comment:
Why are you asking them to "please check your access"? It implies that we aren't confident that it will fix things. Perhaps changing that to "you can confirm your access to the gateway endpoints from an internal EC2 instance" or something along those lines. Still bringing up that *they* may want to confirm things, but that we are confident that they'll get the expected result.